app/waf/src/controllers/cert.controller.ts

```diff
@@ -0,0 +1,270 @@
+    });
+
+    if (!adc.management.trustedDeviceId) {
+      throw new HttpErrors.UnprocessableEntity(`Adc: ${adc.id} is not trusted`);
+    }
+
+    let certContent = certObj!.content || null;
+    let name = certObj!.name || 'f5cert';
+
+    let length = 0;
+    if (certContent !== null) {
+      length = certContent.length;
+    }
+
+    if (!certContent || length === 0) {
+      throw new HttpErrors.UnprocessableEntity('empty cert content');
+    }
+
+    try {
+      await this.asgMgr.uploadFile(
+        adc.management.trustedDeviceId!,
+        certContent,
+        length,
+        name,
+      );
+
+      let remoteName = certObj.name + cert.id;
+      let body = {
+        command: 'install',
+        name: `${remoteName}`,
+        'from-local-file': `/var/config/rest/downloads/${path.basename(
+          cert.localfile,
+        )}`,
+      };
+      await this.asgMgr.installCert(adc.management.trustedDeviceId!, body);
+      cert.installed = true;
+      cert.remotepath = `/Common/${remoteName}`;
+      await this.certRepository.updateById(id, cert);
+    } catch (error) {
+      throw new HttpErrors.UnprocessableEntity(
+        'upload cert to asg service failed',
+      );
+    }
+  }
+
+  @get(prefix + '/certs/{id}/adcs/{adcId}', {
+    responses: {
+      '200': 'Status of cert',
+      '404': Schema.notFound('Can not find relative resource'),
+      '422': Schema.unprocessableEntity(' Adc is not trusted'),
+    },
+  })
+  async certCheckStatus(
+    @param(Schema.pathParameter('id', 'Cert resource ID')) id: string,
+    @param(Schema.pathParameter('adcId', 'ADC resource ID')) adcId: string,
+  ): Promise<Response> {
+    let cert: Cert = await this.certRepository.findById(id, {
+      where: {
+        and: [
+          {or: [{tenantId: await this.tenantId}, {public: true}]},
+          {id: id},
+        ],
+      },
+    });
+    if (cert.length === 0) {
+      throw new EntityNotFoundError(cert.name, cert.id);
+    } else {
+      return new Response(Cert, cert);
+    }
+  }
+
+  @patch(prefix + '/certs/{id}', {
+    responses: {
+      '204': Schema.emptyResponse('Successfully update cert resource'),
+      '404': Schema.notFound('Can not find ADC resource'),
+    },
+  })
+  async updateById(
+    @param(Schema.pathParameter('id', 'Cert resource ID')) id: string,
+    @requestBody(
+      Schema.updateRequest(
+        Cert,
+        'Cert resource properties that need to be updated',
+      ),
+    )
+    cert: Partial<Cert>,
+  ): Promise<void> {
+    await this.certRepository.updateById(id, cert, {
+      tenantId: await this.tenantId,
+    });
+  }
+
+  @del(prefix + '/certs/{id}', {
+    responses: {
+      '204': Schema.emptyResponse('Successfully delete Cert resource'),
+      '404': Schema.notFound('Can not find Cert resource'),
+    },
+  })
+  async deleteById(
+    @param(Schema.pathParameter('id', 'Cert resource ID')) id: string,
+  ): Promise<void> {
+    await this.certRepository.deleteById(id, {
```

Comment:
Maybe we can support removing cert from BIG-IP in the future.

Reply:
This needs the other APIs and we will implement this later. For this one only delete the cert from local db.